src/solvers/dgsem_tree/indicators_1d.jl

```diff
@@ -201,6 +201,40 @@ function (indicator_max::IndicatorMax)(u::AbstractArray{<:Any,3},
   return alpha
 end
 
+function create_cache(::Type{IndicatorClamp}, equations::AbstractEquations{1}, basis::LobattoLegendreBasis)
+
+  alpha = Vector{real(basis)}()
+
+  return (; alpha, indicator_threaded, basis.weights)
+end
+
+function create_cache(typ::Type{IndicatorClamp}, mesh, equations::AbstractEquations{1}, dg::DGSEM, cache)
+  cache = create_cache(typ, equations, dg.basis)
+end
+
+function (indicator_clamp::IndicatorClamp)(u::AbstractArray{<:Any,4},
+                                           mesh, equations, dg::DGSEM, cache;
+                                           kwargs...)
+  @unpack alpha, weights = indicator_clamp.cache
+  resize!(alpha, nelements(dg, cache))
+
+  @threaded for element in eachelement(dg, cache)
+    mean::Variable = 0.0
+    for i in eachnode(dg)
+      u_local = get_node_vars(u, equations, dg, i, element)
+      mean += indicator_clamp.variable(u_local, equations) * weights[i]*0.5
+    end
```

Comment:
It may be worth considering to multiply by `0.5` at the end to use FMAs when computing the mean.

Reply:
Done.